### PR TITLE
Deploy stargate

### PR DESCRIPTION
**Pull request by pmckelvy1**

### CODE REVIEW

File: test/plugins/individual-collateral/stargate/StargateUSDbCTestSuite.test.ts [hunk 1]
```diff
@@ -0,0 +1,278 @@
+import collateralTests from '../collateralTests'
```
Comment (julianmrodri) [inline test/plugins/individual-collateral/stargate/StargateUSDbCTestSuite.test.ts:1]: Is it necessary to duplicate this whole file just ot add `targetNetwork`? Seems like an overkill. Cant it work with the same file for both networks?

File: test/plugins/individual-collateral/stargate/constants.ts [hunk 2]
```diff
@@ -1,25 +1,50 @@
 import { bn, fp } from '#/common/numbers'
 import { networkConfig } from '#/common/configuration'
+import hre from 'hardhat'
```
Comment (julianmrodri) [inline test/plugins/individual-collateral/stargate/constants.ts:3]: Is this import used?

File: test/plugins/individual-collateral/stargate/constants.ts [hunk 3]
```diff
@@ -1,25 +1,50 @@
...
+export const USDT = networkConfig[chainId].tokens['USDT']!
+export const USDC_HOLDER = chainId == '8453' ? '0x4c80e24119cfb836cdf0a6b53dc23f04f7e652ca' : '0x0a59649758aa4d66e25f08dd01271e891fe52199'
+export const USDC_USD_PRICE_FEED = networkConfig[chainId].chainlinkFeeds['USDC']! // currently same key for USDC and USDbC
+export const ETH_USD_PRICE_FEED = networkConfig[chainId].chainlinkFeeds['ETH']!
 export const SUSDC_POOL_ID = bn('1')
 export const WSUSDC_NAME = 'Wrapped S*USDC'
 export const WSUSDC_SYMBOL = 'wS*USDC'
 
+export const USDbC = networkConfig[chainId].tokens['USDbC']!
+export const SUSDbC = networkConfig[chainId].tokens['sUSDbC']!
```
Comment (julianmrodri) [inline test/plugins/individual-collateral/stargate/constants.ts:39]: These two variables are not needed as thy are covered by. `USDC` and `SUSDC` I think? Which already are dynamic based on the network.

File: scripts/deployment/phase2-assets/collaterals/deploy_stargate_usdc_collateral.ts [hunk 4]
```diff
@@ -103,8 +106,8 @@ async function main() {
 
   console.log(`Deployed Stargate ${USDC_NAME} to ${hre.network.name} (${chainIdKey}): ${collateral.address}`)
 
-  assetCollDeployments.collateral.sUSDC = collateral.address
-  assetCollDeployments.erc20s.sUSDC = erc20.address
+  assetCollDeployments.collateral.sgUSDC = collateral.address
```
Comment (julianmrodri) [inline scripts/deployment/phase2-assets/collaterals/deploy_stargate_usdc_collateral.ts:109]: I think this should be `sgUSDbc` right?

File: scripts/addresses/base-3.0.0/8453-tmp-assets-collateral.json [hunk 5]
```diff
@@ -8,7 +8,8 @@
     "USDbC": "0x6490D66B17A1E9a460Ab54131165C8F921aCcDeB",
     "cbETH": "0x5fE248625aC2AB0e17A115fef288f17AF1952402",
     "cUSDbCv3": "0xa372EC846131FBf9AE8b589efa3D041D9a94dF41",
-    "aBasUSDbC": "0x1DdB7dfdC5D26FE1f2aD02d9972f12481346Ae9b"
+    "aBasUSDbC": "0x1DdB7dfdC5D26FE1f2aD02d9972f12481346Ae9b",
+    "sgUSDC": "0x15395aCCbF8c6b28671fe41624D599624709a2D6"
```
Comment (julianmrodri) [inline scripts/addresses/base-3.0.0/8453-tmp-assets-collateral.json:12]: Should this be `sgUSDbC`?

File: scripts/verification/collateral-plugins/verify_stargate_usdbc.ts [hunk 6]
```diff
@@ -0,0 +1,114 @@
...
+      [
+        {
+            priceTimeout: priceTimeout.toString(),
+            chainlinkFeed: networkConfig['8453'].chainlinkFeeds.USDC!,
+            oracleError: oracleError.toString(),
+            erc20: deployments.erc20s.sgUSDC!,
+            maxTradeVolume: fp('1e6').toString(), // $1m,
+            oracleTimeout: oracleTimeout('8453', '86400').toString(), // 24h hr,
+            targetName: hre.ethers.utils.formatBytes32String('USD'),
+            defaultThreshold: fp('0.01').add(oracleError).toString(), // ~2.5%
```
Comment (julianmrodri) [inline scripts/verification/collateral-plugins/verify_stargate_usdbc.ts:101]: I think the comment of ~2,5% is off right?

File: common/configuration.ts [hunk 7]
```diff
@@ -62,10 +62,13 @@ export interface ITokens {
   cbETH?: string
   STG?: string
   sUSDC?: string
+  sUSDbC?: string
```
Comment (julianmrodri) [inline common/configuration.ts:65]: So what is the difference between `sUSDC` `sUSDbC` and `sgUSDC` and `sgUSDbC`? Arent they the same? if thats the case we should only keep one I think

File: scripts/verification/collateral-plugins/verify_stargate_usdbc.ts [hunk 8]
```diff
@@ -0,0 +1,114 @@
+import hre from 'hardhat'
```
Comment (julianmrodri) [inline scripts/verification/collateral-plugins/verify_stargate_usdbc.ts:1]: I think this file should be named verify_stargate_usdc as it works for both right? To match the one in the deployment script

File: common/configuration.ts (hunk 7 above)
Comment (pmckelvy1) [reply common/configuration.ts:65]: sUDSC is the stargate pool, sgUSDC is our wrapped version

Comment (pmckelvy1) [reply common/configuration.ts:65]: updated to wsgUSDC / wsgUSDbC

**Comment by tbrent** [comment]
I went through the deployed contracts on Base and examined each function, and it looks correct to me.

**Pull request merged**